src/components/Checkout/KivaCreditGuestPayment.vue

```diff
@@ -0,0 +1,186 @@
+		</div>
+		<kv-button
+			type="submit"
+			id="kiva-credit-submit"
+			class="tw-mb-2"
+			v-kv-track-event="['payment.continueBtn']"
+			title="Checkout using your Kiva credit"
+			@click.native.prevent="validateCreditBasket"
+		>
+			<slot>Complete order</slot>
+		</kv-button>
+	</div>
+</template>
+
+<script>
+import checkoutUtils from '@/plugins/checkout-utils-mixin';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+import KvCheckbox from '~/@kiva/kv-components/vue/KvCheckbox';
+import KvTextInput from '~/@kiva/kv-components/vue/KvTextInput';
+
+import { validationMixin } from 'vuelidate';
+import { required, email } from 'vuelidate/lib/validators';
+
+export default {
+	name: 'KivaCreditPayment',
+	components: {
+		KvButton,
+		KvCheckbox,
+		KvTextInput
+	},
+	inject: ['apollo', 'cookieStore'],
+	mixins: [
+		checkoutUtils,
+		validationMixin
+	],
+	props: {
+		managedAccountId: {
+			type: String,
+			default: '',
+		},
+		promoFundId: {
+			type: String,
+			default: ''
+		},
+		isGuestCheckout: {
+			type: Boolean,
+			default: false,
+		},
+		promoGuestCheckoutEnabled: {
+			type: Boolean,
+			default: false
+		}
+	},
+	data() {
+		return {
+			email: null,
+			termsAgreement: false,
+			emailUpdates: false,
+		};
+	},
+	validations: {
+		email: {
+			required,
+			email,
+		},
+		termsAgreement: { required: value => value === true },
+	},
+	methods: {
+		validateCreditBasket() {
+			this.$kvTrackEvent('basket', 'Kiva Checkout', 'Button Click');
+			this.$emit('updating-totals', true);
+			this.validateGuestPromoBasket(this.email, this.emailUpdates, this.promoFundId, this.managedAccountId)
+				.then(validationStatus => {
+					if (validationStatus === true) {
+						// succesful validation
+						this.checkoutCreditBasket();
+					} else {
+						// validation failed
+						this.$emit('updating-totals', false);
+						this.showCheckoutError(validationStatus);
+						this.$emit('refreshtotals');
+					}
+				}).catch(errorResponse => {
+					this.$emit('updating-totals', false);
+					console.error(errorResponse);
+				});
+		},
+		checkoutCreditBasket() {
+			this.checkoutBasket(this.promoGuestCheckoutEnabled)
+				.then(transactionResult => {
+					if (typeof transactionResult !== 'object') {
+						// succesful validation
+						this.$kvTrackEvent('basket', 'Kiva Checkout', 'Success', transactionResult);
+						// Complete transaction handles additional analytics + redirect
+						this.$emit('complete-transaction', transactionResult);
+					} else {
+						// checkout failed
+						this.$emit('updating-totals', false);
+						const errorResult = transactionResult?.errors ?? [];
+						this.showCheckoutError(errorResult);
+						this.$emit('checkout-failure', errorResult);
+					}
+				}).catch(errorResponse => {
+					this.$emit('updating-totals', false);
+					console.error(errorResponse);
+				});
+		},
+		validateCampaignEmail() {
+			const emailRegex = new RegExp(`^[A-Za-z0-9._%+-]@${promoFundEmail}`);
```

Comment:
We'll need to remove the `promoFundEmail` variable here and use a regex that validates for any email address instead of a partner specific domain.